source/read/count-documents.txt

```diff
@@ -0,0 +1,232 @@
+.. _java-rs-count:
+
+===============
+Count Documents
+===============
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+.. facet::
+   :name: genre
+   :values: reference
+
+.. meta::
+   :keywords: number, amount, estimation, code example
+
+Overview
+---------
+
+In this guide, you can learn how to retrieve an exact and an estimated count of the
+number of documents in a collection.
+
+Sample Data
+~~~~~~~~~~~
+
+The examples in this guide use the ``sample_restaurants.restaurants`` collection
+from the :atlas:`Atlas sample datasets </sample-data>`. To learn how to create a
+free MongoDB Atlas cluster and load the sample datasets, see
+:ref:`<java-rs-getting-started>`.
+
+.. include:: includes/reactor-note.rst
+
+.. _java-rs-accurate-count:
+
+Retrieve an Exact Count
+-----------------------
+
+Use the ``countDocuments()`` method to count the number of documents in a
+collection. To count the number of documents that match specific search
+criteria, pass a query filter to the ``countDocuments()`` method.
+
+To learn more about specifying a query, see :ref:`java-rs-specify-query`.
+
+Count All Documents
+~~~~~~~~~~~~~~~~~~~
+
+To return a count of all documents in the collection, call the ``countDocuments()``
+method and pass in no parameters, as shown in the following example:
+
+.. code-block:: java
+
+   Publisher<Long> countPublisher = restaurants.countDocuments();
+   Flux.from(countPublisher).doOnNext(System.out::println).blockLast();
```

Comment:
As `countPublisher` will only return a single result its more idiomatic to use a `Mono` - which represents this kind of publisher, that produces at most 1 result. A `Flux` represents a potentially boundless publisher.

So I recommend where only a single result is expected prefer `Mono.from`